packages/webview_flutter/android/src/main/java/io/flutter/plugins/webviewflutter/InputAwareWebView.java

```diff
@@ -112,4 +112,32 @@ public void run() {
         });
     return super.checkInputConnectionProxy(view);
   }
+
+  /**
+   * Ensure that input creation happens back on {@link #containerView}'s thread once this view no
+   * longer has focus.
+   *
+   * <p>The logic in {@link #checkInputConnectionProxy} forces input creation to happen on Webview's
+   * thread for all connections. We undo it here so users will be able to go back to typing in
+   * Flutter UIs as expected.
+   */
+  @Override
+  public void clearFocus() {
+    super.clearFocus();
+    if (proxyAdapterView == null) {
+      // No need to reset the InputConnection to the default thread if we've never changed it.
+      return;
+    }
+    containerView.requestFocus();
+    containerView.post(
+        new Runnable() {
+          @Override
+          public void run() {
+            containerView.onWindowFocusChanged(true);
```

Comment:
[nit, can definitely happen in a different PR]
We're repeating this magic sequence in more than one place, this specific call site isn't document so someone stepping into this code will probably be confused. Would probably be useful to extract these steps to a common method.

Reply:
I'll clean it up in flutter/plugins#1853.